### ISSUE TITLE
failed to compile cargo-update v0.8.1

### ISSUE THREAD

**Issue by cdetrio** [nabijaczleweli/cargo-update#28]
Just to track occurrences of https://users.rust-lang.org/t/crates-io-compatibility-with-old-cargo/9127/23

See also https://github.com/rust-lang/crates.io/issues/580

```
$ cargo install --verbose cargo-update
    Updating registry `https://github.com/rust-lang/crates.io-index`
failed to compile `cargo-update v0.8.1`, intermediate artifacts can be found at `/home/ubuntu/target-install`

Caused by:
  failed to parse registry's information for: serde_derive

Caused by:
  the given version requirement is invalid
```

```
$ cargo -V
cargo 0.8.0 (built 2016-03-22)
$ rustc -V
rustc 1.7.0
```

**Comment by nabijaczleweli** [comment]
I'm deriving from the links that this is unfixable on both end- and dev-user side and we have to wait for external resolution, or...?

**Comment by pravic** [comment]
... or just update rust/cargo - 1.7.0 looks quiet old.

**Comment by nabijaczleweli** [comment]
Good point. Does that change anything, @cdetrio?

**Comment by beatgammit** [comment]
I had an issue compiling 1.0 on Rust 1.15, but updating to 1.17 fixed it (didn't try 1.16), so updating rust/cargo is likely the solution.

**Comment by nabijaczleweli** [comment]
Alrighty, then
